docker-app/qfieldcloud/core/serializers.py

```diff
@@ -535,3 +539,44 @@ class LatestPackageSerializer(serializers.Serializer):
     package_id = serializers.UUIDField()
     packaged_at = serializers.DateTimeField()
     data_last_updated_at = serializers.DateTimeField()
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    organization = serializers.StringRelatedField(source="team_organization")
+
+    class Meta:
+        model = Team
+        fields = ("username", "organization")
+
+
+class TeamMemberSerializer(serializers.ModelSerializer):
+    member = serializers.CharField()
+
+    def to_internal_value(self, data):
+        validated_data = super().to_internal_value(data)
+        email_or_username = data.get("member")
+
+        request_method = self.context["request"].method
+
+        if request_method in ["PUT", "PATCH"]:
+            if not email_or_username:
+                raise serializers.ValidationError(
+                    {"member": "Username must be provided"}
+                )
+
+            if User.objects.fast_count(email_or_username) > 0:
+                raise serializers.ValidationError(
+                    {
+                        "member": f"A user with username '{email_or_username}' already exists."
+                    }
+                )
```

Comment:
```suggestion
        try:
            member = User.objects.fast_search(email_or_username) > 0:
        except User.DoesNotExistError:
            raise serializers.ValidationError(
                    {
                        "member": f"A user with username '{email_or_username}' already exists."
                    }
                )
```

Actually there is no need for a new method, that's why this is throwing.

Reply:
I will check it!

Reply:
It does not work. 

This 'if' checks via fast_search if a user with the new username exists already and therefore prevent another user to get this new username.

So, if we pass a non-existing username (which means that our user can change his/her username), the fast_search will raise an exception as it makes a get request to the db and does not find this username.

Reply:
@tdrivas not sure if I follow.

> So, if we pass a non-existing username (which means that our user can change his/her username), the fast_search will raise an exception as it makes a get request to the db and does not find this username.

Indeed, this is the reason why I suggest wrapping the `fast_search` in a `try/except` block?

I see that the suggestion I gave has a typo in there, check the updated version.

Reply:
You are correct, it does work now. Thanks.